kt/kt.go

```diff
@@ -206,7 +208,56 @@ func NewConnTLS(host string, port int, poolsize int, timeout time.Duration, cred
 
 // NewConn creates a connection to an Kyoto Tycoon endpoint.
 func NewConn(host string, port int, poolsize int, timeout time.Duration) (*Conn, error) {
-	return newConn(host, port, poolsize, timeout, "")
+	parts := strings.Split(host, "://")
+
+	if len(parts) == 1 {
+		return newConn(host, port, poolsize, timeout, "")
+	} else if len(parts) == 2 {
+		if parts[0] != "unix" {
+			return nil, errors.New("NewConn: Unexpected network")
+		}
+		return newUnixConn(parts[1], poolsize, timeout)
+	} else {
+		return nil, errors.New("NewConn: Wrong parameters")
+	}
+}
+
+type unixDialer struct {
+	net.Dialer
+}
+
+func (d *unixDialer) Dial(network, address string) (net.Conn, error) {
+	// Hackery to remove port 80
+	parts := strings.Split(address, ":")
+	return d.Dialer.Dial("unix", parts[0])
+}
+
+func newUnixConn(socket string, poolsize int, timeout time.Duration) (*Conn, error) {
```

Comment:
Can you make it public, please? Also, can you add creds argument for it too to allow using it for SSL enabled socket? I think I prefer to call this function directly instead of using fake port.